### PR TITLE
Rework the docs

### PR DESCRIPTION
**Pull request by GuilhemN**
Extracted from https://github.com/nelmio/NelmioApiDocBundle/pull/1169

### CODE REVIEW

File: Resources/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+.. code-block:: yaml
+
+    nelmio_api_doc:
+        documentation:
+            info:
+                title: My App
+                description: This is an awesome app!
+                version: 1.0.0
+
+To document your routes, you can use annotations in your controllers::
```
Comment (dbu) [inline Resources/doc/index.rst]: ... you can use the SwaggerPHP annotations and the `Nelmio\ApiDocBundle\Annotation\Model` annotation in your controllers::

File: Resources/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+Use models
+----------
+
+As shown in the example above, the bundle provides the ``@Model`` annotation.
+When you use it, the bundle will deduce your model properties.
+
+.. warning::
+
+    The ``@Model`` annotation acts like a ``@Schema`` annotation. If you nest it with a ``@Schema`` annotation, the bundle will consider that
+    you're documenting an array of models.
```
Comment (dbu) [inline Resources/doc/index.rst]: maybe provide some more examples here? i think this is where a lot of issues asking for help (or reporting a bug that is a user error) come from.

File: Resources/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+In case you prefer using the [Symfony PropertyInfo component](https://symfony.com/doc/current/components/property_info.html) (you
+won't be able to use JMS serialization groups), you can disable JMS serializer
+support in your config:
+
+.. code-block:: yaml
+
+    nelmio_api_doc:
+        models: { use_jms: false }
+
+What's supported?
```
Comment (dbu) [inline Resources/doc/index.rst]: i think this should be above even before the install instructions. it tells the reader if she should even care about this bundle...

File: Resources/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        public function fetchUserRewardsAction(User $user)
+        {
+            // ...
+        }
+    }
+
+Use models
+----------
+
+As shown in the example above, the bundle provides the ``@Model`` annotation.
```
Comment (dbu) [inline Resources/doc/index.rst]: we should explain all options of the model annotation, and what they mean.

File: Resources/doc/index.rst (hunk 1 above)
Comment (dbu) [reply Resources/doc/index.rst]: do we support all swagger php annotations or some not? i think we ignore class level things, right?

File: README.md [hunk 5]
```diff
@@ -19,158 +19,12 @@ for your APIs.
 First, open a command console, enter your project directory and execute the following command to download the latest version of this bundle:
 
 ```
-composer require nelmio/api-doc-bundle dev-master
+composer require nelmio/api-doc-bundle
```
[the remark this answers is not in the document]
Comment (GuilhemN) [reply README.md]: I don't think so, I added it to allow people to install the bundle real quick but if they need more explanations there is a link to the docs just under.

File: README.md [hunk 6]
```diff
@@ -19,158 +19,12 @@ for your APIs.
...
-## What's supported?
-
-This bundle supports _Symfony_ route requirements, PHP annotations,
-[_Swagger-Php_](https://github.com/zircote/swagger-php) annotations,
-[_FOSRestBundle_](https://github.com/FriendsOfSymfony/FOSRestBundle) annotations
-and apps using [_Api-Platform_](https://github.com/api-platform/api-platform).
+## Documentation
 
-For models, it supports the Symfony serializer and the JMS serializer.
+[Read the documentation on symfony.com](https://symfony.com/doc/current/bundles/NelmioApiDocBundle/index.html)
```
[the remark this answers is not in the document]
Comment (GuilhemN) [reply README.md]: Yes, not sure that will be instantly synced though.

@javiereguiluz, will it be possible to re-add the master branch on symfony.com once this PR is merged? :)

File: Resources/doc/index.rst (hunk 1 above)
Comment (GuilhemN) [reply Resources/doc/index.rst]: We only support annotations on controller methods. Annotations at class level are indeed not supported :)

**Comment by GuilhemN** [comment]
Thanks for your review, I normally fixed all your comments :)

> overall i feel like we should aim to provide more examples and possibly even references that list all possible options for things. its a bit tricky as we should not duplicate swagger-php/jms serializer/whatnot documentation.

I think the index must be as simple as possible as it's not necessary for everyone to learn all the options we provide. Like I did for areas, I think we should provide complementary articles at the end of the index to explain parts of the bundle that can be further configured.

File: Resources/doc/index.rst (hunk 3 above)
Comment (GuilhemN) [reply Resources/doc/index.rst]: right, updated :)

File: Resources/doc/index.rst (hunk 2 above)
Comment (GuilhemN) [reply Resources/doc/index.rst]: Added, maybe what I added is a bit too long though?

File: Resources/doc/index.rst (hunk 4 above)
Comment (GuilhemN) [reply Resources/doc/index.rst]: done, I wonder whether this should be done in a separate article though

File: Resources/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,181 @@
+NelmioApiDocBundle
+==================
+
+The **NelmioApiDocBundle** bundle allows you to generate a decent documentation
+for your APIs.
```
[the remark this answers is not in the document]
Comment (GuilhemN) [reply Resources/doc/index.rst]: Much better!

**Comment by GuilhemN** [comment]
Merging so that we can finish https://github.com/nelmio/NelmioApiDocBundle/pull/1169 :)

**Comment by dbu** [comment]
> I think the index must be as simple as possible as it's not necessary for everyone to learn all the options we provide. Like I did for areas, I think we should provide complementary articles at the end of the index to explain parts of the bundle that can be further configured.

agreed!

**Pull request merged**